packages/cardpay-cli/scheduled-payment/execute.ts

```diff
@@ -0,0 +1,134 @@
+      .positional('gasTokenAddress', {
+        type: 'string',
+        description: 'The address of the gas token',
+      })
+      .positional('salt', {
+        type: 'string',
+        description: 'Arbitrary string to make SP unique',
+      })
+      .positional('payAt', {
+        type: 'number',
+        description:
+          'Unix UTC time in seconds that represents the point in time when the payment should be executed. Used for one-time scheduled payments. Should be an empty string when recurringDayOfMonth and recurringUntil are set',
+      })
+      .positional('recurringDayOfMonth', {
+        type: 'number',
+        description:
+          'Day of the month on which the payment will be made recurringly (range: 1-31). Used for recurring scheduled payments. In case the month has less than days than the value provided, the payment will me made on the last day of the month. Should be an empty string when payAt is set.',
+      })
+      .positional('recurringUntil', {
+        type: 'number',
+        description:
+          'Unix UTC time in seconds that represents the point in time when the recurring payment should be stopped. Used for recurring scheduled payments. Should be an empty string when payAt is set',
+      })
+      .positional('gasPrice', {
+        type: 'string',
+        description: 'Gas price (in the smallest units of gas token)',
+      })
+      .option('network', NETWORK_OPTION_ANY);
+  },
+  async handler(args: Arguments) {
+    let {
+      network,
+      moduleAddress,
+      tokenAddress,
+      amount,
+      payeeAddress,
+      fixedUSDFee,
+      feePercentage,
+      executionGas,
+      maxGasPrice,
+      gasTokenAddress,
+      salt,
+      payAt,
+      gasPrice,
+      recurringDayOfMonth,
+      recurringUntil,
+    } = args as unknown as {
+      network: string;
+      moduleAddress: string;
+      tokenAddress: string;
+      amount: string;
+      payeeAddress: string;
+      fixedUSDFee: number;
+      feePercentage: number;
+      executionGas: number;
+      maxGasPrice: string;
+      gasTokenAddress: string;
+      salt: string;
+      payAt: number;
+      gasPrice: string;
+      recurringDayOfMonth?: number | undefined;
+      recurringUntil?: number | undefined;
+    };
+    let { web3, signer } = await getEthereumClients(network, getConnectionType(args));
+    let scheduledPaymentModule = await getSDK('ScheduledPaymentModule', web3, signer);
+    let blockExplorer = await getConstant('blockExplorer', web3);
+
+    console.log(`Execute a scheduled payment ...`);
+    let onTxnHash = (txnHash: string) => console.log(`Transaction hash: ${blockExplorer}/tx/${txnHash}`);
+    await scheduledPaymentModule.executeScheduledPayment(
+      moduleAddress,
+      tokenAddress,
+      amount,
+      payeeAddress,
+      fixedUSDFee,
+      feePercentage,
+      executionGas,
+      maxGasPrice,
+      gasTokenAddress,
+      salt,
+      payAt,
+      gasPrice,
+      recurringDayOfMonth,
```

Comment:
When `recurringDayOfMonth` and `recurringUntil` are provided as empty strings, they are transformed to 0 by yargs right? What are the implications of passing 0s as values for these params into `executeScheduledPayment`? Maybe better to make sure to transform them to null in case they aren't given in the cli command (empty strings)

Reply:
updated